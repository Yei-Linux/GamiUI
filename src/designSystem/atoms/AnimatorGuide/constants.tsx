export const Guiders: any = {
  inlove: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "0px", y: "0px" },
    dimensions: { width: "95px", height: "80px" },
  },
  smart: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "-100px", y: "0px" },
    dimensions: { width: "95px", height: "80px" },
  },
  worried: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "95px", y: "0px" },
    dimensions: { width: "95px", height: "80px" },
  },
  thinking: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "0px", y: "-95px" },
    dimensions: { width: "95px", height: "80px" },
  },
  yawning: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "-100px", y: "-95px" },
    dimensions: { width: "95px", height: "80px" },
  },
  drunk: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "95px", y: "-95px" },
    dimensions: { width: "95px", height: "80px" },
  },
  surprised: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "0px", y: "-190px" },
    dimensions: { width: "95px", height: "80px" },
  },
  upset: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "-100px", y: "-190px" },
    dimensions: { width: "95px", height: "80px" },
  },
  sad: {
    source: "https://i.ibb.co/BwSvrW2/brain-animated.png",
    position: { x: "95px", y: "-190px" },
    dimensions: { width: "95px", height: "80px" },
  },
  chill: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "0px", y: "0px" },
    dimensions: { width: "82px", height: "100px" },
  },
  strong: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-85px", y: "0px" },
    dimensions: { width: "110px", height: "100px" },
  },
  box: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "95px", y: "0px" },
    dimensions: { width: "95px", height: "100px" },
  },
  byciclying: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "0px", y: "-100px" },
    dimensions: { width: "90px", height: "110px" },
  },
  basquet: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-95px", y: "-102px" },
    dimensions: { width: "95px", height: "90px" },
  },
  uleule: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "95px", y: "-102px" },
    dimensions: { width: "95px", height: "90px" },
  },
  run: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "0px", y: "-215px" },
    dimensions: { width: "100px", height: "90px" },
  },
  skater: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-100px", y: "-195px" },
    dimensions: { width: "95px", height: "100px" },
  },
  jumper: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "95px", y: "-195px" },
    dimensions: { width: "95px", height: "100px" },
  },
  singer: {
    source: "https://i.ibb.co/0mTZHhV/brain-animated2.png",
    position: { x: "-100px", y: "-330px" },
    dimensions: { width: "110px", height: "120px" },
  },
};

export const options = {
  type: Object.keys(Guiders),
};
