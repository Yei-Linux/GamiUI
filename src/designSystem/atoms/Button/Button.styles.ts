import { css } from "@emotion/react";
import styled from "@emotion/styled";
import { ButtonType } from "../../../core/domain/types";
import { colorLight } from "../../../styles/theme";
import { twinStyles } from "../../../styles/utilities/twinStyles";

const ButtonTypes = (typeStyle: ButtonType) => css`
  background: ${colorLight.button[typeStyle].bg};
  color: ${colorLight.button[typeStyle].color};
  border: ${colorLight.button[typeStyle].border};
`;

export const ButtonWrapper = styled.button<{
  typeStyle?: ButtonType;
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
  textAlign?: string;
  fontWeight?: string;
}>`
  border: none;
  outline: none;
  cursor: pointer;
  appearance: none;
  padding: 10px;

  ${(props: any) => twinStyles(props)};
  ${(props: any) => ButtonTypes(props.typeStyle)};
`;
