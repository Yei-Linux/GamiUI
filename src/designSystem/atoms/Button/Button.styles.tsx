import styled from '@emotion/styled'
import { mixinFlexVariants } from 'styles/mixins/flex'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'
import {
  BorderType,
  ButtonType,
  FontWeightType,
  HeightType,
  ShadowType,
  TextAlignType,
  WidthType,
} from 'core/domain/types'
import { mixinComponentsTypes } from 'styles/mixins/componentsTypes'
import { spacing } from 'styles/tokens/spacing'

export const Button = styled.button<{
  $variant?: ButtonType
  $border?: BorderType
  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
  $textAlign?: TextAlignType
  $fontWeight?: FontWeightType
}>`
  ${mixinFlexVariants({ justifyContent: 'center', alignItems: 'center' })};

  border: none;
  outline: none;
  appearance: none;

  padding: ${spacing.padding.sm} ${spacing.padding.md};

  &:hover {
    cursor: pointer;
  }

  ${({ $variant }) =>
    mixinComponentsTypes('light', $variant || 'primary', 'button')};

  ${({ $border, $shadow, $fontWeight, $width, $textAlign, $heigth }) =>
    setGenericPropStyles({
      border: $border || 'sm',
      shadow: $shadow || 'md',
      width: $width || 'auto',
      heigth: $heigth || 'auto',
      fontWeight: $fontWeight,
      textAlign: $textAlign,
    })};
`
