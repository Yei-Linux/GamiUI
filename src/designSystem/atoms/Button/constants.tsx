export const options = {
  type: [
    "primary",
    "secondary",
    "tertiary",
    "cuaternary",
    "text",
    "link",
    "success",
    "warning",
    "danger",
    "info",
  ] as const,
};
