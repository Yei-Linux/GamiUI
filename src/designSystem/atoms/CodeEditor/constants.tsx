export const CodeCustomTheme = {
  base: 'vs',
  inherit: false,
  colors: {
    'editor.background': '#1d2432',
    'editorCursor.foreground': '#536383',
    'editor.lineHighlightBackground': '#1d2432',
    'editor.selectionBackground': '#536383',
  },
  rules: [
    { token: '', foreground: '63b7fc', background: '1d2432' },
    { token: 'invalid', foreground: 'f44747' },
    { token: 'emphasis', fontStyle: 'italic' },
    { token: 'strong', fontStyle: 'bold' },

    { token: 'variable', foreground: '1AEBFF' },
    { token: 'variable.parameter', foreground: '9CDCFE' },
    { token: 'constant', foreground: 'ff636f' },
    { token: 'comment', foreground: '608B4E' },
    { token: 'number', foreground: '63b7fc' },
    { token: 'regexp', foreground: 'C0C0C0' },
    { token: 'annotation', foreground: 'ff636f' },
    { token: 'type', foreground: '3DC9B0' },

    { token: 'delimiter', foreground: 'dcdcdc' },
    { token: 'delimiter.html', foreground: 'dcdcdc' },

    { token: 'tag', foreground: 'ff636f' },
    { token: 'tag.id.pug', foreground: '4F76AC' },
    { token: 'tag.class.pug', foreground: '4F76AC' },
    { token: 'meta', foreground: 'D4D4D4' },
    { token: 'meta.tag', foreground: '9ddf6e' },
    { token: 'metatag', foreground: 'ff636f' },
    { token: 'metatag.content.html', foreground: '1AEBFF' },
    { token: 'metatag.html', foreground: 'ff636f' },
    { token: 'metatag.xml', foreground: 'ff636f' },
    { token: 'metatag.php', fontStyle: 'bold' },

    { token: 'key', foreground: '9CDCFE' },
    { token: 'string.key', foreground: '9CDCFE' },
    { token: 'string.value', foreground: '9ddf6e' },

    { token: 'attribute.name', foreground: 'ff636f' },
    { token: 'attribute.value', foreground: '3FF23F' },

    { token: 'string', foreground: '9ddf6e' },
    { token: 'string.sql', foreground: 'FF0000' },

    { token: 'keyword', foreground: 'ff636f' },
    { token: 'keyword.flow', foreground: 'C586C0' },

    { token: 'operator.sql', foreground: '778899' },
    { token: 'operator.swift', foreground: '909090' },
    { token: 'predefined.sql', foreground: 'FF00FF' },
  ],
}

export const codeOptions = {
  acceptSuggestionOnCommitCharacter: true,
  acceptSuggestionOnEnter: 'on',
  accessibilitySupport: 'auto',
  autoIndent: false,
  automaticLayout: true,
  codeLens: true,
  colorDecorators: true,
  contextmenu: true,
  cursorBlinking: 'blink',
  cursorSmoothCaretAnimation: false,
  cursorStyle: 'line',
  disableLayerHinting: false,
  disableMonospaceOptimizations: false,
  dragAndDrop: false,
  fixedOverflowWidgets: false,
  folding: true,
  foldingStrategy: 'auto',
  fontLigatures: false,
  formatOnPaste: false,
  formatOnType: false,
  hideCursorInOverviewRuler: false,
  highlightActiveIndentGuide: true,
  links: true,
  mouseWheelZoom: false,
  multiCursorMergeOverlapping: true,
  multiCursorModifier: 'alt',
  overviewRulerBorder: true,
  overviewRulerLanes: 2,
  quickSuggestions: true,
  quickSuggestionsDelay: 100,
  readOnly: false,
  renderControlCharacters: false,
  renderFinalNewline: true,
  renderIndentGuides: true,
  renderLineHighlight: 'all',
  renderWhitespace: 'none',
  revealHorizontalRightPadding: 30,
  roundedSelection: true,
  rulers: [],
  scrollBeyondLastColumn: 5,
  scrollBeyondLastLine: true,
  selectOnLineNumbers: true,
  selectionClipboard: true,
  selectionHighlight: true,
  showFoldingControls: 'mouseover',
  smoothScrolling: false,
  suggestOnTriggerCharacters: true,
  wordBasedSuggestions: true,
  wordSeparators: `~!@#$%^&*()-=+[{]}|;:'",.<>/?`,
  wordWrap: 'off',
  wordWrapBreakAfterCharacters: '\t})]?|&,;',
  wordWrapBreakBeforeCharacters: '{([+',
  wordWrapBreakObtrusiveCharacters: '.',
  wordWrapColumn: 80,
  wordWrapMinified: true,
  wrappingIndent: 'none',
}

export const options = {
  type: [] as const,
}
