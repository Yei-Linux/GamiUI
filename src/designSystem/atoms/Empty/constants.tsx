export const iconsEmpty = {
  balloon: 'drawkit__empty',
  sad: 'sad__multicolor',
}

export const options = {
  type: ['balloon', 'sad'] as const,
}
