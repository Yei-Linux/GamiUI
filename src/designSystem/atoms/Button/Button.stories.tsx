import { Meta } from "@storybook/react";
import {
  getListTemplate,
  stylesControl,
} from "../../../core/helpers/storybook";

import Button from ".";
import { options as optionsButton } from "./constants";
import { options as optionsStyles } from "../../../core/utils/constants";

const ListTemplate = getListTemplate(Button);

export default {
  title: "Atoms/Button",
  component: Button,
  argTypes: {
    children: { control: "text" },
    type: { control: { type: "select", options: optionsButton.type } },
    ...stylesControl,
  },
} as Meta;

export const TypesButton = ListTemplate.bind({});
TypesButton.args = {
  items: optionsButton.type.map((type) => ({ type })),
  field: 'type',
  children: "Text",
  width: "MEDIUM",
  heigth: "SMALL",
  shadow: "",
  border: "ROUNDED",
};

export const WidthButton = ListTemplate.bind({});
WidthButton.args = {
  items: optionsStyles.width.map((width) => ({ width })),
  field: 'width',
  children: "Text",
  heigth: "SMALL",
  shadow: "",
  border: "ROUNDED",
};
WidthButton.storyName = "Width Sizes";
