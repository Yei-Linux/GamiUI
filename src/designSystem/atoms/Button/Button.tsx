import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { ButtonType } from "../../../core/domain/types";
import { ButtonWrapper } from "./Button.styles";

export interface IButton extends IGeneralProps {
  /**
   * Content Button to show
   */
  children: React.ReactNode;
  /**
   * Button Type
   */
  type: ButtonType;
}

const Button = ({ children, type, ...args }: IButton) => {
  return (
    <ButtonWrapper {...args} typeStyle={type} type="submit">
      {children}
    </ButtonWrapper>
  );
};

Button.defaultProps = {
  type: "primary",
  border: "MEDIUM",
  heigth: "SMALL",
  width: "SMALL",
};

export default Button;
