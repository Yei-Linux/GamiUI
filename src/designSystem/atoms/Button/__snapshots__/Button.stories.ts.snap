// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Button Types Button 1`] = `
Array [
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        primary
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-r0r5pw-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        secondary
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-czun26-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        tertiary
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-3zre3d-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        cuaternary
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-1slklvl-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        text
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-4ukf5l-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        link
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-4ukf5l-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        success
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-myz7n3-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        warning
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-16ov8di-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        danger
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-idn3m0-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        info
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-19jr0hz-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
]
`;

exports[`Storyshots Atoms/Button Width Sizes 1`] = `
Array [
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        NONE
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-1laelah-ButtonWrapper ekfoxpm0"
        type="submit"
        width="NONE"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        SMALL
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-139zgec-ButtonWrapper ekfoxpm0"
        type="submit"
        width="SMALL"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        NORMAL
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-egiayo-ButtonWrapper ekfoxpm0"
        type="submit"
        width="NORMAL"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        MEDIUM
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-r0r5pw-ButtonWrapper ekfoxpm0"
        type="submit"
        width="MEDIUM"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        LARGE
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-h91d55-ButtonWrapper ekfoxpm0"
        type="submit"
        width="LARGE"
      >
        Text
      </button>
    </div>
  </div>,
  <div
    className="css-1pnd932-WrapperStories e1kw3nsx0"
  >
    <div
      className="css-12sqrlu-TypeColumn-FirstColumn e1kw3nsx2"
    >
      <p>
        FULL
      </p>
    </div>
    <div
      className="css-qtyufk-TypeColumn-SecondColumn e1kw3nsx3"
    >
      <button
        className="css-1fzx7yv-ButtonWrapper ekfoxpm0"
        type="submit"
        width="FULL"
      >
        Text
      </button>
    </div>
  </div>,
]
`;
